docs/topics/patterns/token_id_webhook.md

```diff
@@ -0,0 +1,361 @@
+            },
+        });
+
+        // 👇 Setup lambda url
+        const lambdaUrl = webhookHandlerLambda.addFunctionUrl({
+            authType: lambda.FunctionUrlAuthType.NONE,
+        });
+
+        apiKeySecret.grantRead(webhookHandlerLambda);
+        apiKeySecret.grantRead(serviceLambda);
+        secretStringSecret.grantRead(webhookHandlerLambda);
+
+        new CfnOutput(this, 'FunctionUrl ', { value: lambdaUrl.url });
+    }
+}
+
+```
+
+4. Add code to the webhook to process incoming messages. Below is sample code for the webhook lambda handler that extracts
+a user's `token_id` from the webhook payload, and access resources stored in Momento cache. It also verifies that the webhook caller
+is indeed Momento through the signing secret.
+
+```typescript
+
+import {GetSecretValueCommand, SecretsManagerClient} from '@aws-sdk/client-secrets-manager';
+import {CacheClient, CacheGet, CacheListPushFront, Configurations, CredentialProvider} from '@gomomento/sdk';
+import * as crypto from "crypto";
+
+const _secretsClient = new SecretsManagerClient({});
+const _cachedSecrets = new Map<string, string>();
+let _cacheClient: CacheClient | undefined = undefined;
+
+const cacheName = 'course-comments';
+
+export const handler = async (event: any) => {
+  try {
+
+    const secretStringSecretName = process.env.THE_SIGNING_SECRET;
+
+    if (secretStringSecretName === undefined) {
+      throw new Error("Missing required env var 'THE_SIGNING_SECRET");
+    }
+
+    const secretString = await getSecret(secretStringSecretName);
+    const authorized = verifySignature(event.headers['momento-signature'], secretString, event.body);
+    if (!authorized) {
+      return {
+        statusCode: 403,
+        headers: {
+          'Content-Type': 'application/json',
+          'Access-Control-Allow-Origin': '*',
+        },
+        body: '{"message": "Access Denied!"}',
+      };
+    }
+
+    const payload = JSON.parse(event.body)
+    const userID = payload.token_id
+    const message = JSON.parse(payload.text)
+
+    console.log('Storing user comment for userID ' + userID + ' and courseId ' + message.courseId);
+    console.log('Comment: ' + message.comment);
+
+    const cacheClient = await getCacheClient();
+    if (_cacheClient === undefined) {
+      throw new Error('Cache client is undefined');
+    }
+
+    const listResp = await cacheClient.listPushFront(cacheName, String(message.courseId),
+      JSON.stringify({userID: userID, comment: message.comment}));
+    if (listResp instanceof CacheListPushFront.Success) {
+      console.log('Successfully persisted comment for course');
+    } else if (listResp instanceof CacheListPushFront.Error) {
+      console.log('Error while publishing comment for course ' + listResp.message());
+    }
+
+    return {
+      statusCode: 200,
+      headers: {
+        'Content-Type': 'application/json',
+        'Access-Control-Allow-Origin': '*',
```

Comment:
You don't need the `Access-Control-Allow-Origin` header because this endpoint isn't being called from a browser.